From 064b3a476fd1a84ad009ce0a151930f82b67d49c Mon Sep 17 00:00:00 2001
From: Ky-Anh Huynh <kyanh@theslinux.org>
Date: Sat, 20 Apr 2013 17:11:18 +0700
Subject: [PATCH] pkgbase will define the behavior

As "pkgbase" is a string, while "pkgname" may be an array,
we can only get "pkgbase" from environment.
---
 pacman/src/pacman-4.1.0/scripts/makepkg.sh.in | 12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

diff --git a/pacman/src/pacman-4.1.0/scripts/makepkg.sh.in b/pacman/src/pacman-4.1.0/scripts/makepkg.sh.in
index b887819..b1eb313 100644
--- a/pacman/src/pacman-4.1.0/scripts/makepkg.sh.in
+++ b/pacman/src/pacman-4.1.0/scripts/makepkg.sh.in
@@ -2733,7 +2733,6 @@ unset pkgname pkgbase pkgver pkgrel epoch pkgdesc url license groups provides
 unset md5sums replaces depends conflicts backup source install changelog build
 unset makedepends optdepends options noextract
 
-pkgname="${PACKAGE_NAME:-}"
 pkgver="${PACKAGE_VERSION:-}"
 pkgrel="${PACKAGE_RELEASE:-}"
 pkgbase="${PACKAGE_BASE:-}"
@@ -2765,6 +2764,17 @@ else
 	source_safe "$BUILDFILE"
 fi
 
+# There are some cases
+#   PACKAGE_BASE is unset:
+#     `pkgname`, `pkgbase` will be from `PKGBUILD`
+#
+#   PACKAGE_BASE is set:
+#     `pkgbase` is defined
+#     `pkgname` will be `pkgbase` if `pkgname` is unset
+#
+if [[ -z "$pkgname" ]]; then
+	pkgname="$pkgbase"
+fi
 # set defaults if they weren't specified in buildfile
 pkgbase=${pkgbase:-${pkgname[0]}}
 epoch=${epoch:-0}
-- 
1.8.3

